immutable_tree.go

```diff
@@ -152,21 +152,41 @@ func (t *ImmutableTree) Get(key []byte) (index int64, value []byte) {
 	if t.root == nil {
 		return 0, nil
 	}
+	return t.root.get(t, key)
+}
 
+// GetFast returns the value of the specified key if it exists, or nil. 
+// The returned value must not be modified, since it may point to data stored within IAVL.
+// GetFast employs a more performant strategy than Get for retrieving the value.
+func (t *ImmutableTree) GetFast(key []byte) []byte {
```

Comment:
This `GetFast` method is created because one of the [proofs](https://github.com/osmosis-labs/iavl/blob/roman/fast-node-get-set/proof_ics23.go#L33) and many unit tests use an index returned by the regular `Get` method. The fast nodes do not have the index since they are not in the tree.

The plan is to create a new issue/PR that will deprecate the index. At which point, the `Get` and `GetFast` methods can be combined together. There needs to be an issue on the SDK side to convert to the new method signature since this is a breaking change.

The original suggestion to do this was made [here](https://github.com/osmosis-labs/iavl/issues/1#issuecomment-1011602632)